src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/validation/AccountValidator.java

```diff
@@ -27,5 +27,7 @@ public AccountValidator(RestRequest request, BytesReference ref, Settings esSett
         super(request, ref, esSettings, param);
         allowedKeys.put("current_password", DataType.STRING);
         mandatoryKeys.add("current_password");
+        mandatoryOrKeys.add("hash");
```

Comment:
I did consider this. Looks like InternalUsersValidator also inherits from the same and for that it is optional as some of the internal user apis are designed that way. Putting it there would make it mandatory for internal user apis. 
Hence I put it in AccountValidator

Reply:
I would prefer to remove the check especially that it is present in `AccountApiAction` with probably a better error message if the validation does not pass.

Reply:
This seems to consistent with the other APIs. If the mandatory keys are absent, then it throws a bad request with invalid configuration error.

Reply:
Why is it necessary to validate it twice? If you want to rely on `AccountValidator`, remove the check from `AccountApiAction` (line 217). I'd prefer the check to be done in `AccountApiAction`.

Reply:
If you pass 
{
"password": "",
"hash": ""
}
then the validator check wont fail as the params exist. 
But it fails in the account api action because they are empty/null

Reply:
OK, so `AccountValidator` can't even invalidate all cases.